gcp_variant_transforms/transforms/vcf_field_conflict_resolver.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import vcf
+
+__all__ = ['FieldConflictResolver']
+
+class FieldConflictResolver(object):
+  """A class for resolving all VCF field related mistmatches."""
+
+  def __init__(self, split_alternate_allele_info_fields=True):
+    # type: (bool) -> None
+    """Initialize the class.
+
+    Args:
+     split_alternate_allele_info_fields (bool): Whether INFO fields with
+       `Number=A` are store under the alternate_bases record.
+     """
+    self._split_alternate_allele_info_fields = (
+        split_alternate_allele_info_fields)
+
+  def resolve(self, vcf_field_key, first_vcf_field_value,
+              second_vcf_field_value):
+    # type: (str, Union[str, int], Union[str, int]) -> Union[str, int]
+
+    """Returns resolution for the conflicting field values.
+
+    Args:
+      vcf_field_key: field key in VCF header.
+      first_vcf_field_value: first field value.
+      second_vcf_field_value: second field value.
+    Raises:
+      ValueError: if the conflict cannot be resolved.
+    """
+    if vcf_field_key == 'type':
+      return self._resolve_type(first_vcf_field_value, second_vcf_field_value)
+    elif vcf_field_key == 'num':
+      return self._resolve_number(first_vcf_field_value, second_vcf_field_value)
+    else:
+      # We only care about conflicts in 'num' and 'type' fields.
+      return first_vcf_field_value
+
+  def _resolve_type(self, first, second):
+    if first == second:
+      return first
+    elif (first in ('Integer', 'Float') and second in ('Integer', 'Float')):
```

Comment:
Consider making 'Integer' and 'Float' constants.

Reply:
Done!